client/payment-request/event-handlers.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * Internal dependencies
+ */
+import {
+	normalizeShippingAddress,
+	normalizeOrderData,
+	getErrorMessageFromNotice,
+} from './utils';
+
+export const shippingAddressChangeHandler = async ( api, event ) => {
+	const response = await api.paymentRequestCalculateShippingOptions(
+		normalizeShippingAddress( event.shippingAddress )
+	);
+
+	// Possible statuses success, fail, invalid_payer_name, invalid_payer_email, invalid_payer_phone, invalid_shipping_address.
+	event.updateWith( {
+		status: response.result,
+		shippingOptions: response.shipping_options,
+		total: response.total,
+		displayItems: response.displayItems,
+	} );
+};
+
+export const shippingOptionChangeHandler = async ( api, event ) => {
+	const response = await api.paymentRequestUpdateShippingDetails( event );
+
+	if ( 'success' === response.result ) {
+		event.updateWith( {
+			status: 'success',
+			total: response.total,
+			displayItems: response.displayItems,
+		} );
+	}
+
+	if ( 'fail' === response.result ) {
+		event.updateWith( { status: 'fail' } );
+	}
+};
+
+export const paymentMethodHandler = async (
+	api,
+	completePayment,
+	abortPayment,
+	event
+) => {
+	// Kick off checkout processing step.
+	const response = await api.paymentRequestCreateOrder(
+		normalizeOrderData( event )
+	);
+
+	if ( 'success' === response.result ) {
+		try {
+			const confirmation = api.confirmIntent( response.redirect );
+			// We need to call `complete` outside of `completePayment` to close the dialog for 3DS.
+			event.complete( 'success' );
+
+			// `true` means there is no intent to confirm.
+			if ( true === confirmation ) {
```

Comment:
This was taken from `confirm-card-payment.js` [here](https://github.com/Automattic/woocommerce-payments/blob/0ece4e0791971a08e622134ff60d0e417e083392/client/checkout/blocks/confirm-card-payment.js#L17), from the credit card checkout.

> Actually — after reading the comment — I'm thinking willNotConfirm might be a better name? Or intentWasRejected?

I didn't understand the reasoning behind calling it the opposite name. Can you explain?

It kinda makes sense to call it `confirmation` or `confirmed` because `api.confirmIntent` might return `true` at first, but then it can return a `confirmation` Promise, which then resolves to a redirect URL.

Reply:
> it can return a confirmation Promise, which then resolves to a redirect URL.

Ahhh, in that case I think we should keep the `true === confirmation` check. And it also makes sense to keep the `confirmation` variable name.

>I didn't understand the reasoning behind calling it the opposite name. Can you explain?

No 😅 

I don't know why it's called the opposite. Maybe the blame will reveal who worked on this last? I doubt they remember the reasoning though 😬